Python/Product/PythonTools/PythonTools/Project/ConfigureTestFrameworkInfoBar.cs

```diff
@@ -112,16 +112,21 @@ internal abstract class TestFrameworkInfoBar : PythonInfoBar {
         private async Task<bool> InstallPyTestAsync() {
             var packageManagers = _infoBarData.InterpreterOptionsService.GetPackageManagers(_infoBarData.InterpreterFactory);
 
-            foreach (var packageManager in packageManagers) {
-                bool pytestInstalled = await packageManager.InstallAsync(
-                    new PackageSpec("pytest"),
-                    new VsPackageManagerUI(Site),
-                    CancellationToken.None
-                );
-
-                if (pytestInstalled) {
-                    return true;
+            try {
+                foreach (var packageManager in packageManagers) {
+                    bool pytestInstalled = await packageManager.InstallAsync(
```

Comment:
do we want the catch inside the loop, so that it still continues?

Reply:
Yeah why not. It exits on the first one that succeeds. It's more complicated logic though. If it's the only package manager, it needs to output the error. Otherwise succeed. I think I can just save the exception in that case.